src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java

```diff
@@ -0,0 +1,222 @@
+        Repository repo = pr.getToRef().getRepository();
+
+        // compat with Stash < 3.2.0
+        String repoPath = this.properties.getRepositoryDir(repo).getAbsolutePath();
+
+        Settings settings = context.getSettings();
+        List<String> exe = new LinkedList<String>();
+        exe.add(this.getExecutable(
+            settings.getString("exe"),
+            settings.getBoolean("safe_path", false)).getPath());
+
+        if (settings.getString("params") != null) {
+            for (String arg : settings.getString("params").split("\r\n")) {
+                exe.add(arg);
+            }
+        }
+
+        StashUser currentUser = authCtx.getCurrentUser();
+        ProcessBuilder pb = new ProcessBuilder(exe);
+
+        Map<String, String> env = pb.environment();
+        env.put("STASH_USER_NAME", currentUser.getName());
+        if (currentUser.getEmailAddress() != null) {
+            env.put("STASH_USER_EMAIL", currentUser.getEmailAddress());
+        } else {
+            log.error("Can't get user email address. getEmailAddress() call returns null");
+        }
+        env.put("STASH_REPO_NAME", repo.getName());
+
+        boolean isAdmin = permissions.hasRepositoryPermission(
+            currentUser, repo, Permission.REPO_ADMIN);
+        env.put("STASH_IS_ADMIN", String.valueOf(isAdmin));
+
+        RepositoryCloneLinksRequest.Builder cloneLinksRequestBuilder =
+            new RepositoryCloneLinksRequest.Builder();
+
+        cloneLinksRequestBuilder.repository(repo);
+
+        RepositoryCloneLinksRequest cloneLinksRequest =
+            cloneLinksRequestBuilder.build();
+
+        Set<NamedLink> cloneLinks = this.repoService.getCloneLinks(
+            cloneLinksRequest
+        );
+
+        for (NamedLink link : cloneLinks) {
+            env.put(
+                "STASH_REPO_CLONE_" + link.getName().toUpperCase(),
+                link.getHref()
+            );
+        }
+
+        env.put(
+            "STASH_BASE_URL",
+            this.properties.getBaseUrl().toString()
+        );
+
+        env.put("STASH_PROJECT_NAME", repo.getProject().getName());
+        env.put("STASH_PROJECT_KEY", repo.getProject().getKey());
+
+        String summaryMsg = "Merge request failed";
+
+        pb.directory(new File(repoPath));
+        pb.redirectErrorStream(true);
+        try {
+            Process process = pb.start();
+            InputStreamReader input = new InputStreamReader(
+                process.getInputStream(), "UTF-8");
+            OutputStream output = process.getOutputStream();
+
+            output.write((
+                          pr.getFromRef().getRepository().getSlug() + " " +
+                          pr.getFromRef().getLatestChangeset() + " " +
+                          pr.getToRef().getRepository().getSlug() + " " +
+                          pr.getToRef().getLatestChangeset() + "\n"
+                          ).getBytes("UTF-8"));
+            output.close();
```

Comment:
I thought about this some more last night and I'm fairly convinced that the first three things on stdin should be the information you describe.  That way we can share scripts among hooks and any script not interested in data beyond those three pieces of information can ignore it.  This could break scripts that read via a loop and/or don't expect information more than those three things on stdin but I think that's probably easy enough to fix.

For all of the information available in the pull request, would it be useful to put a stream of XML or JSON on stdin after the "standard" input?  It seems kind of clunky to me to try to use environment variables or individual input fields on stdin to describe the information because it is structured.

What do you think?